### DISCUSSION TITLE
[Unknown Mod][Heroes] The hero stops upon discovery

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#8998] [Ideas]
One option that I would like to have in the game is:
"The hero stops moving when he reveals a point of interest hidden under the fog of war."

Start 0Test14.MP2.
Select a destination eight movement points east.

![The Hero Stops Upon Discovery 01](https://user-images.githubusercontent.com/43583994/90168051-dcf67f00-dd9c-11ea-8767-6d0963783a9f.png)

Click to accept the default path.
The blue hero will move to the destination without stopping and position himself near the opposing green hero.

![The Hero Stops Upon Discovery 02](https://user-images.githubusercontent.com/43583994/90168151-fd263e00-dd9c-11ea-989b-04918cb78fdb.png)

Now with the option enabled, as soon as the green enemy hero is revealed, the blue hero stops his progress and waits for a new order.

![The Hero Stops Upon Discovery 03](https://user-images.githubusercontent.com/43583994/90168233-16c78580-dd9d-11ea-891d-d919db58b1c7.png)

The blue player can then decide to send his hero south, away from the enemy green hero.
He then selects a destination eight movement points south.

![The Hero Stops Upon Discovery 04](https://user-images.githubusercontent.com/43583994/90168342-3eb6e900-dd9d-11ea-800c-d7f51ce25b01.png)

As soon as the ore mine is revealed, the blue hero stops his progress and waits for a new order.

![The Hero Stops Upon Discovery 05](https://user-images.githubusercontent.com/43583994/90168422-58f0c700-dd9d-11ea-8308-2d2546e28b1d.png)

This operation can be applied to all kinds of adventure map objects, opposing heroes, mines, towns, etc.
The player can validate only the points of interest that he wishes in an adjustment file (example an .xml file).

Files: https://drive.google.com/file/d/1ew019aLIOy5Nc_UqSSsTJ-Ek0AVDzmFR/view?usp=sharing

You can read the discussion on this here:
https://heroes2.forumactif.com/t939-beyond-1-0-the-hero-stops-upon-discovery#8446

**Comment by ihhub** [comment]
Moving this to a discussion section. None of HoMM games have and it kills all the purpose of making mistakes by human players which is a part of the gameplay.
